axum/src/docs/extract.md

```diff
@@ -604,27 +653,20 @@ async fn auth_middleware<B>(
 where
     B: Send,
 {
-    // running extractors requires a `RequestParts`
-    let mut request_parts = RequestParts::new(request);
+    // running extractors requires a `axum::http::request::Parts`
+    let (mut parts, body) = request.into_parts();
 
-    // `TypedHeader<Authorization<Bearer>>` extracts the auth token but
-    // `RequestParts::extract` works with anything that implements `FromRequest`
-    let auth = request_parts.extract::<TypedHeader<Authorization<Bearer>>>()
+    // `TypedHeader<Authorization<Bearer>>` extracts the auth token
+    let auth = TypedHeader::<Authorization<Bearer>>::from_request_parts(&mut parts, &())
```

Comment:
I think we should provide `RequestExt { fn extract(self) -> _; fn extract_parts(&mut self) -> _; }` and `PartsExt { fn extract(&mut self) -> _; }` or sth. like that, such that people can use method call syntax, and also don't need to do the whole `into_parts` + `from_parts` dance, to extract headers and such from a request.

Reply:
Yep I agree. My plan was to add those when the big PR is merged.